tests/commands/test_analyze.py

```diff
@@ -0,0 +1,292 @@
+    args = Namespace(filepath=fake_charm, force=None)
+    AnalyzeCommand("group", config).run(args)
+
+    expected_titles = ["Attributes:", "Lint Errors:"]
+    logged = [rec.message for rec in caplog.records]
+    assert all(x in logged for x in expected_titles)
+
+
+def test_complete_set_of_results(caplog, config, monkeypatch, tmp_path):
+    """Show a complete basic case of results."""
+    caplog.set_level(logging.DEBUG, logger="charmcraft")
+
+    # fake results from the analyzer
+    linting_results = [
+        linters.CheckResult(
+            name="check-lint-01",
+            check_type=linters.CheckType.lint,
+            url="url-01",
+            text="text-01",
+            result=linters.WARNINGS,
+        ),
+        linters.CheckResult(
+            name="check-lint-02",
+            check_type=linters.CheckType.lint,
+            url="url-02",
+            text="text-02",
+            result=linters.OK,
+        ),
+        linters.CheckResult(
+            name="check-lint-03",
+            check_type=linters.CheckType.lint,
+            url="url-03",
+            text="text-03",
+            result=linters.ERRORS,
+        ),
+        linters.CheckResult(
+            name="check-attribute-04",
+            check_type=linters.CheckType.attribute,
+            url="url-04",
+            text="text-04",
+            result="check-result-04",
+        ),
+        linters.CheckResult(
+            name="check-attribute-05",
+            check_type=linters.CheckType.attribute,
+            url="url-05",
+            text="text-05",
+            result=linters.IGNORED,
+        ),
+        linters.CheckResult(
+            name="check-lint-06",
+            check_type=linters.CheckType.lint,
+            url="url-06",
+            text="text-06",
+            result=linters.IGNORED,
+        ),
+        linters.CheckResult(
+            name="check-lint-07",
+            check_type=linters.CheckType.lint,
+            url="url-07",
+            text="text-07",
+            result=linters.FATAL,
+        ),
+    ]
+
+    fake_charm = create_a_valid_zip(tmp_path)
+    args = Namespace(filepath=fake_charm, force=None)
+    monkeypatch.setattr(linters, "analyze", lambda *a, **k: linting_results)
+    with patch.object(linters, "analyze") as mock_analyze:
+        mock_analyze.return_value = linting_results
+        AnalyzeCommand("group", config).run(args)
+    mock_analyze.assert_called_with(config, ANY, override_ignore_config=False)
+
+    expected = [
+        "Attributes:",
+        "- check-attribute-04: check-result-04 (url-04)",
+        "- check-attribute-05: ignored (url-05)",
+        "Lint Ignored:",
+        "- check-lint-06 (url-06)",
+        "Lint Warnings:",
+        "- check-lint-01: text-01 (url-01)",
+        "Lint Errors:",
+        "- check-lint-03: text-03 (url-03)",
+        "Lint Fatal:",
+        "- check-lint-07 (url-07)",
+        "Lint OK:",
+        "- check-lint-02: no issues found (url-02)",
+    ]
```

Comment:
@sergiusens please review this sequence of results (from an UX POV), thanks!

Reply:
I'm good with the format